src/routes/common/transactions.helper.ts

```diff
@@ -0,0 +1,176 @@
+ * @param data call data
+ * @returns boolean
+ */
+export const isMultiSendCall = (data: string): boolean => {
+  const MULTISEND_TX_SIGNATURE = 'multiSend(bytes)';
+
+  return isCalldata(data, MULTISEND_TX_SIGNATURE);
+};
+
+interface MultiSendTransactionData {
+  readonly to: string;
+  readonly data: string;
+}
+
+/**
+ * Decodes the transactions contained in a multiSend call
+ * @param encodedMultiSendData multiSend call data
+ * @returns array of individual transaction data
+ */
+export const decodeMultiSendTxs = (
+  encodedMultiSendData: string,
+): MultiSendTransactionData[] => {
+  // uint8 operation, address to, uint256 value, uint256 dataLength
+  const INDIVIDUAL_TX_DATA_LENGTH = 2 + 40 + 64 + 64;
+
+  const MULTISEND_FRAGMENT =
+    'function multiSend(bytes memory transactions) public payable';
+
+  const multiSendInterface = new ethers.Interface([MULTISEND_FRAGMENT]);
+
+  const [decodedMultiSendData] = multiSendInterface.decodeFunctionData(
+    MULTISEND_FRAGMENT,
+    encodedMultiSendData,
+  );
+
+  const txs: MultiSendTransactionData[] = [];
+
+  // Decode after 0x
+  let index = 2;
+
+  while (index < decodedMultiSendData.length) {
+    const txDataEncoded = `0x${decodedMultiSendData.slice(
+      index,
+      // Traverse next transaction
+      (index += INDIVIDUAL_TX_DATA_LENGTH),
+    )}`;
+
+    // Decode operation, to, value, dataLength
+    const [, txTo, , txDataBytesLength] =
+      ethers.AbiCoder.defaultAbiCoder().decode(
+        ['uint8', 'address', 'uint256', 'uint256'],
+        ethers.zeroPadValue(txDataEncoded, 32 * 4),
+      );
+
+    // Each byte is represented by two characters
+    const dataLength = Number(txDataBytesLength) * 2;
+
+    const txData = `0x${decodedMultiSendData.slice(
+      index,
+      // Traverse data length
+      (index += dataLength),
+    )}`;
+
+    txs.push({
+      to: txTo,
+      data: txData,
+    });
+  }
+
+  return txs;
+};
+
+/**
+ * Extracts the common `to` address from a multisend transaction
+ *
+ * @param data multisend call data
+ * @returns the `to` address of all batched transactions contained in `data` or `undefined` if the transactions do not share one common `to` address.
+ */
+export const getSafeAddressFromMultiSend = async (
+  chainId: string,
+  data: string,
+): Promise<string | void> => {
+  const individualTxs = decodeMultiSendTxs(data);
+
+  if (individualTxs.length === 0) {
+    return;
+  }
+
+  const isEveryTxExecTx = individualTxs.every(({ data }) => {
+    return isExecTransactionCall(data);
+  });
+
+  if (!isEveryTxExecTx) {
+    return;
+  }
+
+  const firstRecipient = individualTxs[0].to;
+
+  const isSameToAddress = individualTxs.every(({ to }) => {
+    return to === firstRecipient;
+  });
+
+  if (!isSameToAddress) {
+    return;
+  }
+
+  if (!(await isSafeContract(chainId, firstRecipient))) {
+    return;
+  }
+
+  return firstRecipient;
+};
+
+/**
+ * Validates if a multiSend call is valid for a given Safe address:
```

Comment:
This comment is a bit outdated.
This method does not consider safe addresses. 
```suggestion
 * Validates that the call `data` is `multiSend` and the `to` address is the MultiSendCallOnly deployment.
```